lib/workers/testbot.js

```diff
@@ -0,0 +1,267 @@
+*
+* @returns {String} path to the dev interface
+*/
+
+const getDevInterface = async (devicePath) => {
+  if (process.platform === 'linux') {
+    const devIdLink = '/dev/disk/by-id/usb-Generic_Ultra_HS-SD_MMC_000008264001-0:0'
+    return fs.realpathAsync(devIdLink).catch((error) => {
+      throw new Error(`Could not find testbot. Reason: ${error}`)
+    })
+  }
+
+  if (_.isUndefined(devicePath)) {
+    throw new Error('For non-linux operating systems testbot dev interface has to be provided')
+  }
+
+  return devicePath
+}
+
+module.exports = class TestBotWorker {
+  /**
+  * Represents a TestBotWorker
+  * @constructor
+  * @param {String} title - Name of the worker
+  * @param {String} deviceType - The device type this worker is controlling
+  * @param {Object} [options={}] - Extra options to configure the worker
+  */
+
+  constructor (title, deviceType, options = {}) {
+    this.title = title
+    this.deviceType = deviceType
+    this.options = options
+
+    this.board = new Board(options.devicePath)
+    this.board.serialConfig({
+      portId: HW_SERIAL5,
+      baud: 9600
+    })
+  }
+
+  /**
+  * Send an array of bytes over the selected serial port
+  * @function
+  * @param {Number} command - First byte (i.e. pin to write)
+  * @param {Number} a_ - Second byte
+  * @param {Number} b_ - Third byte
+  */
+
+  sendCommand (command, a_ = 0, b_ = 0) {
+    this.board.serialWrite(HW_SERIAL5, [ command, a_, b_ ])
+  }
+
+  /**
+  * Connected the SD card interface to the target
+  * @function
+  *
+  * @returns Wait time for the firmware to write the serial
+  */
+
+  async switchSdToTarget () {
+    this.sendCommand(SD_RESET_ENABLE)
+
+    // Allow some time to write to serial
+    await Bluebird.delay(10)
+    this.sendCommand(SD_RESET_DISABLE)
+
+    this.board.digitalWrite(LED_PIN, this.board.LOW)
+    this.board.digitalWrite(SD_MUX_SEL_PIN, this.board.LOW)
+
+    return Bluebird.delay(2000)
+  }
+
+  /**
+  * Connected the SD card interface to the host
+  * @function
+  *
+  * @returns Wait time for the firmware to write the serial
+  */
+
+  async switchSdToHost () {
+    this.sendCommand(SD_RESET_ENABLE)
+
+    // Allow some time to write to serial
+    await Bluebird.delay(10)
+    this.sendCommand(SD_RESET_DISABLE)
+
+    this.board.digitalWrite(LED_PIN, this.board.HIGH)
+    this.board.digitalWrite(SD_MUX_SEL_PIN, this.board.HIGH)
+
+    return Bluebird.delay(2000)
+  }
+
+  /**
+  * Power target board on
+  * @function
+  */
+
+  powerTarget () {
+    this.sendCommand(ENABLE_VOUT_SW)
+  }
+
+  /**
+  * Power target board off
+  * @function
+  */
+
+  poweroffTarget () {
+    this.sendCommand(DISABLE_VOUT_SW)
+  }
+
+  /**
+  * Flash SD card with image
+  * @function
+  * @param {String} - path to the image
+  */
+
+  async flash (image) {
+    await this.switchSdToHost()
+
+    // For linux, udev will provide us with a nice id for the testbot
+    const destination = await getDevInterface(this.options.devicePath)
```

Comment:
Can we do something like this instead?

```js
const deviceInterface = process.platform === 'linux' ? DEVICE_ID_LINK : this.options.devicePath

try {
    const destination = await fs.realpathAsync(deviceInterface)
} catch (error) {
    throw new Error(`Could not find testbot. Reason: ${error}`)
}
```

And then we simplify the confusing `getDevInterface` function that doesn't use its arguments on the Linux branch?

Reply:
Forgot about this, fixed now.